### PR TITLE
Update style docs

### PR DESCRIPTION
**Pull request by timhoffm**

### CODE REVIEW

File: doc/api/style_api.rst [hunk 1]
```diff
@@ -2,20 +2,33 @@
...
+The ``style`` module adds support for easy-to-switch plotting "styles" with the 
+same parameters as a matplotlib rc file that is read at startup to configure 
+matplotlib.
 
-   Examples of using style sheets with :func:`matplotlib.style.use` can be found at :doc:`/gallery/style_sheets/style_sheets_reference`.
+:doc:`/tutorials/introductory/customizing` describes the mechanism and usage
+of styles.
+
+The :doc:`/gallery/style_sheets/style_sheets_reference` gives an overview of
+the builtin styles.
```
Comment (jklymak) [inline doc/api/style_api.rst]: I added the pre-amble above, and removed the "see also".  I agree that the bulk of the documentation can be in the tutorial, but I think its helpful to say what the module does in a sentence too before sending the user elsewhere.

Comment (jklymak) [reply doc/api/style_api.rst]: BTW, hope it was OK to push to your branch.  And feel free to revert if you disagree....

Comment (timhoffm) [reply doc/api/style_api.rst]: That's ok (even though I find that wording not particularly great).

File: doc/api/style_api.rst [hunk 2]
```diff
@@ -2,20 +2,33 @@
 ``matplotlib.style``
 ********************
 
-.. seealso::
+The ``style`` module adds support for easy-to-switch plotting "styles" with the 
+same parameters as a matplotlib rc file that is read at startup to configure 
+matplotlib.
```
Comment (QuLogic) [inline doc/api/style_api.rst]:
```suggestion
same parameters as a :file:`matplotlibrc` file that is read at startup to
configure Matplotlib.
```

File: lib/matplotlib/style/core.py [hunk 3]
```diff
@@ -70,6 +70,11 @@ def use(style):
     The style name of 'default' is reserved for reverting back to
     the default style settings.
 
+    .. note::
+
+       This updates the ``rcParams`` with the settings from the style.
+       ``rcParams`` not defined in the style are kept.
```
Comment (anntzer) [inline lib/matplotlib/style/core.py]:
```
`.rcParams`
```
will link correctly.

**Comment by timhoffm** [comment]
First sentence reworded.

**Pull request merged**